packages/mui-material/src/Slider/Slider.js

```diff
@@ -89,44 +58,111 @@ export const SliderRoot = styled('span', {
       transition: 'none',
     },
   },
+  variants: [
+    ...Object.keys((theme.vars ?? theme).palette)
+      .filter((key) => (theme.vars ?? theme).palette[key].main)
+      .map((color) => ({
+        props: { color },
+        style: {
+          color: (theme.vars || theme).palette[color].main,
+        },
+      })),
```

Comment:
Another regression occurred since Material UI v5.15.14 - #41772. This evaluation fails for nested fields, when the `color` here becomes an object rather than a string.

---

Side note: We could consider using `reduce()` here to avoid two separate loops.